app/models/manageiq/providers/lenovo/physical_infra_manager/event_catcher/stream.rb

```diff
@@ -44,29 +47,45 @@ def parse_events(events)
     end
 
     # Update the @last_event_ems_ref with the new last ems_ref if to exist new events
-    @last_event_ems_ref = parsed_events.last[:ems_ref]
+    @last_event_ems_ref = parsed_events.last[:ems_ref].to_i
     parsed_events
   end
 
   def events
     expression = { :filterType => 'FIELDNOTREGEXAND', :fields => filter_fields }
-
     opts = {'filterWith' => expression.to_json}
 
+    @last_cn ||= get_last_cn(opts)
+    opts['headers'] = pagination unless @last_event_ems_ref >= @last_cn
+
     connection.fetch_events(opts)
   end
 
+  def get_last_cn(opts)
+    headers = { :range => 'item=0-1' }
+    opts['headers'] = headers
+    connection.get_last_cn(opts) # get last event cn based on the response
+  end
+
+  def pagination
```

Comment:
Also a comment about this routine would be welcome. Mainly explaining this calculation `@limit = @last_cn * 0.07 + offset`.

If this `0.07` could be a variable with a more semantical name, the method understanding would be improved